packages/@azure/servicebus/data-plane/lib/queueClient.ts

```diff
@@ -122,47 +117,43 @@ export class QueueClient extends Client {
-   * - For sending a message to a `session` enabled Queue, please set the `sessionId` property of
-   * the message.
-   * - For sending a message to a `partition` enabled Queue, please set the `partitionKey` property
-   * of the message.
-   * For more information please see {@link https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-partitioning#use-of-partition-keys Use of partition keys}
+   * To send a message to a `session` or `partition` enabled Queue, please set the
+   * `sessionId` property and `partitionKey` properties respectively.
    *
-   * @param data - Message to send.  Will be sent as UTF8-encoded JSON string.
-   * @returns Promise<Delivery>
+   * @param data - Message to send.
+   * @returns Promise<void>
    */
-  async send(data: SendableMessageInfo): Promise<Delivery> {
+  async send(data: SendableMessageInfo): Promise<void> {
     const sender = MessageSender.create(this._context);
     return sender.send(data);
   }
 
   /**
-   * Sends a batch of SendableMessageInfo to the ServiceBus Queue. The "message_annotations",
-   * "application_properties" and "properties" of the first message will be set as that of
-   * the envelope (batch message).
-   * - For sending a message to a `session` enabled Queue, please set the `sessionId` property of
-   * the message.
-   * - For sending a message to a `partition` enabled Queue, please set the `partitionKey` property
-   * of the message.
-   * For more information please see {@link https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-partitioning#use-of-partition-keys Use of partition keys}
+   * Sends a batch of SendableMessageInfo to the ServiceBus Queue in a single AMQP message.
+   * To send messages to a `session` or `partition` enabled Queue, please set the
+   * `sessionId` property and `partitionKey` properties respectively.
+   *
+   * All messages in the batch should have the same `sessionId` and `partitionKey` properties.
+   * If sending to a `partition` enabled Queue, `partitionKey` is mandatory.
    *
    * @param datas - An array of SendableMessageInfo objects to be sent in a Batch message.
-   * @return Promise<Delivery>
+   * @return Promise<void>
    */
-  async sendBatch(datas: SendableMessageInfo[]): Promise<Delivery> {
+  async sendBatch(datas: SendableMessageInfo[]): Promise<void> {
     const sender = MessageSender.create(this._context);
     return sender.sendBatch(datas);
   }
 
   /**
-   * Starts the receiver by establishing an AMQP session and an AMQP receiver link on the session.
-   * Messages will be passed to the provided onMessage handler and error will be passed to the
-   * provided onError handler.
+   * Starts the receiver in streaming mode by establishing an AMQP session and an AMQP receiver
+   * link on the session.
+   *
+   * @param onMessage - Callback for each incoming message.
+   * @param onError - Callback for any error that occurs while receiving messages.
```

Comment:
I like the updated documentation, clearer!
I'd update it a little more
onMessage - "Callback for processing each incoming message."
onError - "Callback that gets executed for any error that occurs during receiving or processing of messages."